routes/searchMenu.js

```diff
@@ -7,53 +7,108 @@ const Menu = require('../models/menu.js');
             }
         });
-        const menuTags = await TaggedMenu.findAll({
+
+        //Find coordinates
+        const restaurantInfo = await Restaurant.findOne({
+            attributes: ['Name', 'Address', 'City', 'Country', 'Latitude', 'Longitude'],
+            where: {
+                Restaurant_ID: menuInfo.dataValues.Restaurant_ID
+            }
+        });
+        let menuTags = await TaggedMenu.findAll({
             attributes: ['Tag'],
             where: {
                 Menu_ID: req.params.menuID
             }
         });
+
+        // Retrieve the color of the tag
+        menuTags = await menuTags.map( async t => {
+            const tag = await Tag.findOne({
+                where: {
+                    Name: t.dataValues.Tag
+                }
+            });
+            return {
+                name: tag.dataValues.Name,
+                color: tag.dataValues.Color
+            }
+        });
+        const tags = await Promise.all(menuTags);
+
         const menuItemsWithoutTags= await MenuItem.findAll({
             where: {
                 Menu_ID: req.params.menuID
             }
         });
-        const tags = await menuTags.map( m => { return m.dataValues.Tag });
 
-        const promises =  menuItemsWithoutTags.map(async menuItems => {
-            const tagsOnItem = await TaggedItem.findAll({
+        const promises = menuItemsWithoutTags.map(async menuItems => {
+            let tagsOnItem = await TaggedItem.findAll({
                 attributes: ['Tag'],
                 where: {
                     MI_ID: menuItems.dataValues.MI_ID
                 }
             });
-            const tags = await tagsOnItem.map( m => { return m.dataValues.Tag });
 
+            tagsOnItem = await tagsOnItem.map( async t => {
+                const tag = await Tag.findOne({
+                    where: {
+                        Name: t.dataValues.Tag
+                    }
+                });
+                return {
+                    name: tag.dataValues.Name,
+                    color: tag.dataValues.Color
+                }
+            });
+
+            const ratings = await ItemReview.findAll({
+                attributes: ['Rating'],
+                where: {
+                    MI_ID: menuItems.dataValues.MI_ID
+                }
+            });
+            const average = await aggregateRating(ratings);
+            const tags = await Promise.all(tagsOnItem);
 
             return {
                 name: menuItems.dataValues.Name,
                 description: menuItems.dataValues.Description,
+                type: menuItems.dataValues.Type,
                 priceEuros: menuItems.dataValues.Price,
                 imageLink: menuItems.dataValues.ImageLink,
-                tags}
+                rating: average,
+                tags
+            }
         });
         const menuItems = await Promise.all(promises);
 
         res.status(200).send({
```

Comment:
Maybe I'm just blind, but I can find the response I will get if no menuItems are found?

Reply:
You are right, now you just get an empty response together with the 400 status. Will fix it